packages/react-devtools-shared/src/backend/utils.js

```diff
@@ -275,3 +275,26 @@ export function isSynchronousXHRSupported(): boolean {
     window.document.featurePolicy.allowsFeature('sync-xhr')
   );
 }
+
+// https://www.npmjs.com/package/semver-compare
+export function semvercmp(a: string = '', b: string = ''): number {
+  const pa = a.split('.');
+  const pb = b.split('.');
+  for (let i = 0; i < 3; i++) {
+    const na = +pa[i];
+    const nb = +pb[i];
```

Comment:
This function assumes all characters in the string are numbers after split by `'.'`. However in renderer.js there a usage of `gte(version, '18.0.0-alpha')`

Reply:
Hmm. Good point. This appears to convert to `NaN`.  

I'd have to think about the intended semantics here a bit more to decide on what an appropriate numeric conversion and comparison would be.  Let me get back to you on that one.

Reply:
I found this https://semver.org/#spec-item-11
Hope it's helpful!
BTW I'm fine with a simplified compare logic as long as it works with React version numbers

Reply:
Nice catch!

Reply:
So it turns out that the `NaN` behavior here actually works as desired.

I pulled in the test code from https://unpkg.com/browse/semver-compare@1.0.0/test/cmp.js and expanded it with several pre-release examples.  Those do actually sort correctly, like:

```ts
      expect(sortedVersions).toEqual([
        '1.2.3',
        '1.5.5',
        '1.5.19',
        '2.3.1',
        '4.1.3',
        '4.2.0',
        '4.11.6',
        '10.5.5',
        // Specifically check non-numerical pre-release versions
        '11.0.0-alpha.0',
        '11.0.0-alpha.1',
        '11.0.0',
        '11.3.0',
      ]);
```

I also tweaked the logic to handle more dots.  I don't think React uses that specific `-alpha.N` convention the way I do for Redux, but it was easy enough to add.

I think this resolves that issue.

Reply:
Could you also check that it doesn't break behavior for versions for `@next` (e.g. `18.3.0-next-758fc7fde-20230207` and `@experimental` (e.g. `0.0.0-experimental-758fc7fde-20230207`) release channels? Probably best to test at least [every current NPM tag](https://www.npmjs.com/package/react?activeTab=versions)

Reply:
consider this case
```js
const unsortedVersions = ['18.0.2', '18.0.1', '18.0.1-alpha', '18.0.0-alpha.1', '18.0.2-alpha', '18.0.1-alpha.1', '18.0.0'];
unsortedVersions.slice().sort(semvercmp);
(7) ['18.0.1-alpha', '18.0.2-alpha', '18.0.0-alpha.1', '18.0.1-alpha.1', '18.0.0', '18.0.1', '18.0.2']
```

Reply:
I did some additional searching on NPM.  I'd originally been copying code from https://www.npmjs.com/package/semver-compare .  After looking around, I found https://www.npmjs.com/package/compare-versions . It's a bit larger (~200 lines, but that includes lots of comments, and it's only 625b min+gz), but seems to 100% match results from using the actual `semver` library.

I can either add `compare-versions` as a dependency, or copy-paste and inline it with attribution as I was doing earlier.  Is there a preference which we go with?

Reply:
I vote for `compare-versions`.
Thanks for doing this!

Reply:
Awesome! thank you all for sorting this all out 🙇🏼